test/studies/ml/lib/Chai.chpl

```diff
@@ -0,0 +1,828 @@
+                // const dL_dB = dL_dY;
+
+                // if dL_dX.shape != input.shape then tn.err("Dense backwardBatch: dL_dX.shape != input.shape");
+                // if dL_dW.shape != weights.shape then tn.err("Dense backwardBatch: dL_dW.shape != weights.shape");
+                // if dL_dB.shape != bias.shape then tn.err("Dense backwardBatch: dL_dB.shape != bias.shape");
+
+                // newDeltas[i] = dL_dX;
+                // biasGradient += dL_dY.data;
+                // weightsGradient += dL_dW.data;
+            }
+            this.biasGrad.data += biasGradient;
+            this.weightsGrad.data += weightsGradient;
+            return newDeltas;
+        }
+
+        proc ref optimize(mag: real(64)) {
+            this.bias -= mag * this.biasGrad;
+            this.weights -= mag * this.weightsGrad;
+        }
+        proc ref resetGradients() {
+            this.biasGrad.data = 0;
+            this.weightsGrad.data = 0;
+        }
+
+        proc write(fw: IO.fileWriter) throws {
+            bias.write(fw);
+            weights.write(fw);
+        }
+        proc read(fr: IO.fileReader) throws {
+            bias.read(fr);
+            weights.read(fr);
+            uninitialized = false;
+        }
+        proc signature(): string {
+            return "Dense(" + outputSize + ")";
+        }
+    }
+
+    record Sigmoid {
+        proc init() { }
+
+        proc forwardProp(x: Tensor(?rank)): Tensor(rank) {
+            const activation = tn.sigmoid(x);
+            return activation;
+        }
+
+        proc forwardPropBatch(batch: [] ?tensorType) where isSubtype(tensorType, Tensor) {
+            // if tensorType <= Tensor then compilerError("Sigmoid forwardPropBatch: tensorType must be a Tensor");
+            // param rank = tensorTypes.rank;
+            // var activations: [batch.domain] batch.eltType;
+            // for i in activations.domain {
+            //     activations[i] = forwardProp(batch[i]);
+            // }
+            // return activations;
+            return [b in batch] forwardProp(b);
+        }
+
+        proc backward(delta: Tensor(?rank),lastInput: Tensor(rank)): Tensor(rank) {
+            const sp = tn.sigmoidPrime(lastInput);
+            const grad = delta * sp;
+            return grad; 
+        }
+
+        proc backwardBatch(deltas: [] ?tensorType1, inputs: [] ?tensorType2) where isSubtype(tensorType1, Tensor) && isSubtype(tensorType2, Tensor) {
+            // var grads: [inputs.domain] inputs.eltType;
+            // for i in grads.domain {
+            //     grads[i] = backward(deltas[i],inputs[i]);
+            // }
+            // return grads;
+            return [(d,i) in zip(deltas,inputs)] backward(d,i);
+        }
+
+        proc optimize(mag: real) { }
+        proc resetGradients() { }
+
+        proc write(fw: IO.fileWriter) throws {
+            // fw.write("[sigmoid]");
+        }
+        proc read(fr: IO.fileReader) throws { }
+
```

Comment:
It looks like the other layer types have a `signature` method, but this one doesn't. Could you add one here.